pkg/injector/patch.go

```diff
@@ -138,41 +139,71 @@ func addContainer(target, add []corev1.Container, basePath string) (patch []JSON
 			path += "/-"
 		}
 		patch = append(patch, JSONPatchOperation{
-			Op:    "add",
+			Op:    addOperation,
 			Path:  path,
 			Value: value,
 		})
 	}
 	return patch
 }
 
-func updateAnnotation(target, add map[string]string, basePath string) (patch []JSONPatchOperation) {
-	for key, value := range add {
-		if target == nil {
-			// First one will be a Create
-			target = map[string]string{}
-			patch = append(patch, JSONPatchOperation{
-				Op:   "add",
-				Path: basePath,
-				Value: map[string]string{
-					key: value,
-				},
-			})
-		} else {
-			// Update
-			op := "add"
-			if target[key] != "" {
-				op = "replace"
-			}
-			patch = append(patch, JSONPatchOperation{
-				Op:    op,
-				Path:  path.Join(basePath, escapeJSONPointerValue(key)),
-				Value: value,
-			})
+// createMapPatchOperation is used specifically for the very first 'add' operation
+// for a non-existent 'path'.
+func createMapPatchOperation(key, path, value string) JSONPatchOperation {
+	return JSONPatchOperation{
+		Op:   addOperation,
+		Path: path,
+		Value: map[string]string{
+			key: value,
+		},
+	}
```

Comment:
I would very much like this to be moved back to how it was, to be consistent with the other patch functions. I see no reason for this to be a separate function.